src/main/groovy/org/pih/warehouse/api/StockMovementStatusContext.groovy

```diff
@@ -0,0 +1,70 @@
+package org.pih.warehouse.api
+
+import org.pih.warehouse.api.StockMovementDirection
+import org.pih.warehouse.auth.AuthService
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.requisition.Requisition
+import org.pih.warehouse.requisition.RequisitionSourceType
+import org.pih.warehouse.shipping.Shipment
+
+class StockMovementStatusContext {
+    Order order
+
+    Shipment shipment
+
+    Requisition requisition
+
+    Location origin
+
+    Location destination
+
+    Shipment getShipment() {
```

Comment:
Are these three getters required (`getShipment`, `getOrder`, `getRequisition`)?

Reply:
I created these 3 getters specifically for tests because otherwise I had some trouble stubbing it.
Do you @EWaterman maybe know if there is any other way for creating a nested stub?

Reply:
You might be able to do something like this:
```
StockMovementStatusContext context = Stub(StockMovementStatusContext) {
        isReturn() >> false
        isInbound() >> true
        ...
}
context.shipment = Stub(Shipment) {
        getStatus() >> new ShipmentStatus(code: ShipmentStatusCode.SHIPPED)
}
context.requisition = new Requisition(status: status)
```

Reply:
tried it and for some reason it doesn't work, `context.shipment` returns null when inspecting it in a debugger :disappointed:

Reply:
Ok, we can leave it as is then